pkg/kubectl/describe.go

```diff
@@ -53,7 +53,13 @@ import (
 // if the output could not be generated. Implementers typically
 // abstract the retrieval of the named object from a remote server.
 type Describer interface {
-	Describe(namespace, name string) (output string, err error)
+	Describe(namespace, name string, describerSetting *DescriberSetting) (output string, err error)
```

Comment:
We avoid pointers in apis unless the absense of a value (nil) is something we care about.